oc-bibtex-actions.el

```diff
@@ -223,6 +226,72 @@ strings by style."
   (when-let ((keys (bibtex-actions-get-key-org-cite)))
     (cons 'oc-citation (bibtex-actions--stringify-keys keys))))
 
+;;; Functions for editing/modifying citations
+
+(defun oc-bibtex-actions-delete-citation ()
+  "Delete the citation-reference or citation at point."
+  (interactive)
+  (org-cite-delete-citation (org-element-context)))
+
+(defun oc-bibtex-actions-cite-swap (i j lst)
+  "Swap index I and J in the list LST."
+  (let ((tempi (nth i lst)))
+    (setf (nth i lst) (nth j lst))
+    (setf (nth j lst) tempi))
+  lst)
+
+(defun oc-bibtex-actions--shift-reference (direction datum)
+  "When point is on a citation-reference DATUM, shift it in DIRECTION."
+  ;; adapted from org-ref-cite
+  (let*  ((current-citation (if (eq 'citation (org-element-type datum)) datum
+                             (org-element-property :parent datum)))
+         (current-ref (when (eq 'citation-reference (org-element-type datum)) datum))
+         (refs (org-cite-get-references current-citation))
+         (index
```

Comment:
Index isn't currently calculated correctly; just nil.

Reply:
@aikrahguzar - if and when you have a moment, can you take a look at this and see if something jumps out that I'm missing?

Reply:
I got this working, but I don't quite understand how. It could be the code in ORC was more complex because of its internal details?